stackrox/stackrox
# Delete obsolete search reducers and sagas for policies in UI

## Description

This follow up task came to mind when I found where to adjust policies path for visit links in global search.

### Analysis

1. Dashboard
    * mapDispatchToProps: no occurrences
    * mapStateToProps: no occurrences
    * useDispatch: no occurrences
    * useSelector: no occurrences

2. Policies
    * getPoliciesSearch: no occurrences (with partial match)
    * setPoliciesSearch: no occurrences (with partial match)

3. Search
    * getGlobalSearch: has occurrences in SearchModal.js and SearchResults.tsx
    * setGlobalSearch: has occurrences in SearchModal.js and SearchResults.tsx

The `takeEveryNewlyMatchedLocation(mainPath, …)` call in searchSagas.js is for global search, not main dashboard.

### Changed files

1. Edit Containers/Search/SearchResults.tsx
    * Delete `passthroughGlobalSearchOptions` action
    * Replace `onLinkHandler` higher-order function which encapsulated obsolete action with `onClick={() => onClose(getLink(item, id))}` prop

2. Edit reducers/globalSearch.js
    * Delete `passthroughGlobalSearchOptions` and so on

3. Edit reducers/policies/reducer.js
    * Delete references to search

4. Delete reducers/policies/search.js

5. Edit sagas/globalSearchSagas.js
    * Delete `watchPassthroughGlobalSearchOptions` call

6. Edit sagas/searchSagas.js
    * Delete `takeEveryNewlyMatchedLocation` call for `policiesPath`

7. Edit sagas/searchSagas.test.js
    * Delete test for policies

### Residue

1. Containers/Search/SearchResults.tsx
    * Replace `onFilterLinkHandler` higher-order function which encapsulated obsolete action with pure function to compute url with search filter
    * Increase type-safety
2. services/SearchService.ts
    * Declare more types which correspond to backend service proto

## Checklist

## Testing Performed

### manual testing

1. `yarn build` in ui
2. `wc build/static/js/*.chunk.js` in ui
    * branch - master: -1538 = 10399007 - 10400545
3. `yarn start` in ui

    * Visit /main/dashboard
        Before changes: see 1 GET /v1/search/metadata/options? request
        After changes: see 1 GET /v1/search/metadata/options? request

    * Visit /main/policy-management/policies
        Before changes: see 2 GET /v1/search/metadata/options?categories=POLICIES requests
        After changes: see 1 GET /v1/search/metadata/options?categories=POLICIES request

    * Visit another container, (global) search for a policy, and then click its visit link

### integration testing

1. `yarn cypress-open` in ui/apps/platform
    * policies/policiesTable.test.js has 3 tests of search filter
